client/state/data-layer/wpcom-api-middleware.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal dependencies
+ */
+import handlers from './wpcom';
+
+/**
+ * WPCOM Middleware API
+ *
+ * Intercepts actions requesting data provided by the
+ * WordPress.com API and passes them off to the
+ * appropriate handler.
+ */
+export const middleware = store => next => action =>
+	// we won't use has( handlers, action.type )
+	// here becuase of performance implications
```

Comment:
Yeah, I ran some tests and object lookup is actually quite a bit faster than Map, at least for my test case. I'd link to jsperf but it's ... very broken. Either way, not that concerned.